benches/thrust/reduce/by_key.cu

```diff
@@ -0,0 +1,70 @@
+#include <nvbench/nvbench.cuh>
+
+#include <thrust/device_vector.h>
+#include <thrust/unique.h>
+#include <thrust/reduce.h>
+#include <thrust/sequence.h>
+
+#include <tbm/range_generator.cuh>
+
+template <typename T>
+thrust::device_vector<T> gen_keys(thrust::device_vector<T> keys)
```

Comment:
Same note about copying.

Seeing how common this is for key algorithms, it might be worthwhile to add a `sorted_random` pattern to the input generator. This can be future work.